commodore/catalog.py

```diff
@@ -124,3 +129,42 @@ def catalog_list(cfg):
             click.echo(f" - {display_name}")
         else:
             click.echo(catalog_id)
+
+
+def list_components(config: Config, cluster_id: str, output: Optional[str] = None):
+    """
+    List all components required by target. Generate list of components
+    by searching for classes with prefix `components.` in the inventory files.
+
+    Component repos are searched in `GLOBAL_GIT_BASE/commodore_components`.
+    """
+
+    try:
+        cluster = load_cluster_from_api(config, cluster_id)
+    except ApiError as e:
+        raise click.ClickException(f"While fetching cluster specification: {e}") from e
+
+    # pylint: disable=import-outside-toplevel
+    import tempfile
+
+    with tempfile.TemporaryDirectory() as tmp:
+        config.inventory.work_dir = P(tmp)
+
+        update_target(config, config.inventory.bootstrap_target)
+        update_params(config.inventory, cluster)
+
+        # Fetch components and config
+        fetch_global_config(config, cluster)
+        fetch_customer_config(config, cluster)
+
+        click.secho("Discovering components...", bold=True)
+        component_names, component_aliases = _discover_components(config)
+        urls, versions = _read_components(config, component_names)
```

Comment:
If you use "private" methods outside the module, please change them to not be "private" anymore.